### ISSUE TITLE
ListBlobs() > Missing mandatory parameters for valid Shared Access Signature

### ISSUE THREAD

**Issue by aL3891** [Azure/azure-storage-net#352]
Hello, 
I'm trying to list the blobs in side a container using a sas token

```
CloudBlobClient blobClient = new CloudBlobClient(new Uri("<token>"));
CloudBlobContainer container = blobClient.GetContainerReference("test");
var b = container.ListBlobs().ToList();
```

I do see request being made in fiddler and it does appear to return the data I expect, but the sdk throws an error

```
An unhandled exception of type 'Microsoft.WindowsAzure.Storage.StorageException' occurred in Microsoft.WindowsAzure.Storage.dll
Additional information: Missing mandatory parameters for valid Shared Access Signature
```

i'm using Microsoft.WindowsAzure.Storage.dll version 7.2.1 from nuget. is this a bug or am I missing something? the sas url also works fine in storage explorer

**Comment by erezvani1529** [comment]
Hi @aL3891,

Would you please share a full stack trace here and mention which version of the library and which target framework you are working with?

Thanks!

**Comment by aL3891** [comment]
Sure thing:

    at Microsoft.WindowsAzure.Storage.Core.Executor.Executor.ExecuteSync[T](RESTCommand`1 cmd, IRetryPolicy policy, OperationContext operationContext) in c:\Program Files (x86)\Jenkins\workspace\release_dotnet_master\Lib\ClassLibraryCommon\Core\Executor\Executor.cs:line 604
    at Microsoft.WindowsAzure.Storage.Blob.CloudBlobContainer.<>c__DisplayClassf.<ListBlobs>b__e(IContinuationToken token) in c:\Program Files (x86)\Jenkins\workspace\release_dotnet_master\Lib\ClassLibraryCommon\Blob\CloudBlobContainer.cs:line 936
    at Microsoft.WindowsAzure.Storage.Core.Util.CommonUtility.<LazyEnumerable>d__0`1.MoveNext() in c:\Program Files (x86)\Jenkins\workspace\release_dotnet_master\Lib\Common\Core\Util\CommonUtility.cs:line 322
    at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
    at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
    at ConsoleApplication2.Program.Main(String[] args) in d:\documents\visual studio 2015\Projects\ConsoleApplication2\ConsoleApplication2\Program.cs:line 13
    at System.AppDomain._nExecuteAssembly(RuntimeAssembly assembly, String[] args)
    at System.AppDomain.ExecuteAssembly(String assemblyFile, Evidence assemblySecurity, String[] args)
    at Microsoft.VisualStudio.HostingProcess.HostProc.RunUsersAssembly()
    at System.Threading.ThreadHelper.ThreadStart_Context(Object state)
    at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
    at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
    at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
    at System.Threading.ThreadHelper.ThreadStart()

The version used is 7.2.1 Microsoft.WindowsAzure.Storage.dll from nuget (WindowsAzure.Storage) and i'm targeting net full framework 4.6.1

**Comment by erezvani1529** [comment]
Thanks @aL3891 for providing the info. We are looking into this issue and will update this thread soon. 

Also, for security reasons, please never share your SAS info publicly (we edited it out for now; you might want to rotate your account key if possible),

**Comment by erezvani1529** [comment]
Hi @aL3891,

Update: This is an Account SAS related bug on our side. We have a fix for it which will be released with 8.0 package. Will close this issue with the fix PR during the release process.

Thanks again for reporting this issue!

**Comment by aL3891** [comment]
Allright, thanks! When will that be coming out do you think? 

This was only a test storage account on my test sub with read permissions, just to give you a repro, but thanks for your concern :)

**Comment by erezvani1529** [comment]
We are planning to release as soon as the new service version is rolled out, before the end of the year.

**Comment by aL3891** [comment]
Allright, cool!
Is there any known work around?

**Comment by erezvani1529** [comment]
Sure,

The workaround would be to provide credentials through creating a StorageCredentials object such as below:

`StorageCredentials accountSAS = new StorageCredentials(/*Your SAS Token*/);
CloudStorageAccount accountWithSAS = new CloudStorageAccount(accountSAS, blobClient.StorageUri, /*default*/null, null, null);
CloudBlobClient blobClientWithSAS = accountWithSAS.CreateCloudBlobClient();
CloudBlobContainer container = blobClient.GetContainerReference(containerName);`

Please let us know if this helps.

Thanks!

**Comment by aL3891** [comment]
That does work, thanks again!

**Comment by erezvani1529** [comment]
Hi @aL3891,

Closing this issue with the change included in latest published package on nuget [v8.0.0](https://www.nuget.org/packages/WindowsAzure.Storage/) 

Thanks!
